R/string_operation.R

```diff
@@ -510,8 +510,14 @@ do_ngram <- function(df, token, sentence, document, maxn=2, sep="_"){
   document_col <- col_name(substitute(document))
 
   # this is executed for ngrams not to be connected over sentences
-  grouped <- df %>%
-    dplyr::group_by(!!!rlang::syms(c(document_col, sentence_col))) # convert the column name to symbol for colum names with backticks
+  grouped <- NULL
+  if (sentence_col != document_col) {
+    grouped <- df %>%
+      dplyr::group_by(!!!rlang::syms(c(document_col, sentence_col))) # convert the column name to symbol for column names with backticks
+  } else { # if document_col and sentence_col are identical, just group the data frame by the document_col.If you group by with identical columns, group_by throws an error.
+    grouped <- df %>%
+      dplyr::group_by(!!!rlang::syms(c(document_col))) # convert the column name to symbol for column names with backticks
```

Comment:
c() does not seem necessary.
rlang::sym seems to be enough since it is handling a single column.

Reply:
Changed it to use ```!!rlang::sym``` as suggested.